src/lib.rs

```diff
@@ -36,26 +36,17 @@ use serde_json::value::Value;
 // Rename type for crate
 type BTreeMap = std::collections::BTreeMap<String, Value>;
 
-pub struct Request {
-    request: hyper::server::Request,
-}
+mod request;
+use request::Request;
```

Comment:
so, this all looks great, but it should be

1. extern crate statements
2. mod statements
3. use statements

rather than interspersing mod and use